lib/AST/GraphOperationInfo.cpp

```diff
@@ -193,3 +196,16 @@ int tf::decodeShapeAttr(const ASTContext &ctx, SymbolicValue attr,
   }
   return arrayValue.size();
 }
+
+/// Return the TF_DataType represented by `value`.
+unsigned tf::getTFDataType(SymbolicValue value) {
+  value = value.lookThroughSingleElementAggregates();
+  assert(value.getKind() == SymbolicValue::Integer);
+  return value.getIntegerValue().sextOrTrunc(32).getLimitedValue();
```

Comment:
should we do some sanity check (assert) on the value validity (range) before returning?

Reply:
I can assert that it's nonzero. Is there a way to assert that it's not too high, other than hardcoding that TF_UINT64 is currently the highest datatype?

Reply:
I don't see a programmatic way, but checking on TF_UINT64 sounds fine -- given TF's backward compatibility guarantee, new types should only get appended to the enum. When that happens, we'll probably need to extend our stdlib to use them, and in that case we can update this enum value range accordingly as well.

Another option is to not check on the upper bound. I'm fine with that too.